starport-core/src/main/java/com/krux/starport/metric/CollectionsUtils.java

```diff
@@ -0,0 +1,16 @@
+package com.krux.starport.metric;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CollectionsUtils {
+
+    public static <T> Collection<List<T>> partition(final Collection<T> wholeCollection, final int partitionSize) {
+        final int[] itemCounter = new int[]{0};
+
+        return wholeCollection.stream()
```

Comment:
@realstraw I think this is the same as Scala's `grouped`. Have I got that right?

Reply:
so why are we not using `grouped`?

Reply:
We can. I just wanted to check my understanding of what the original Java code was doing.